### ISSUE TITLE
Adding different plot types.

### ISSUE THREAD

**Issue by Avamander** [linuxmasterrace/Townvalds#37]
Towny has different plot types but they're fairly useless. But I thought if implemented a bit different they would be really fun and useful.

a) "Arena" plot type:
- Player dies but keeps it's inventory, is teleported to predefined plot in the same plot (or adjacent arena plots if the arena is bigger)
- PvP is enabled between any player and is toggled on when game master starts the game
- Destroying blocks is disabled for everyone who has joined the plot with some special command
- Game master commands, preset player in the same town: (optional, but would enhance the possibility of different minigames a lot)
  - Game master can use certain commands to give certain debuffs/buffs to player(s) for some certain minigames
  - Starting/ending the game
  - Removing/adding people and so on

b) "Private":
- Basically the same as /plot claim
- By default switch by any other town member except mayor is disabled
- User is notified on claim how to restrict building and other types of activities in the plot

c) "Wilds":
- Mob spawns are enabled by default
- Player entering is notified somehow about entering the area

d) "Public":
- No PvP
- For public use things like railway stations and mob farms
- If something like dynmap is added then it's displayed on the dynmap somehow as well.

**Comment by random-nick** [comment]
Not sure about the Arena plot type, that seems very abusable.

**Comment by random-nick** [comment]
https://github.com/linuxmasterrace/Townvalds/issues/37#event-630316160
